foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/descriptors/DescriptorIterator.java

```diff
@@ -707,6 +707,10 @@ public class CascadeCondition{
         public CascadeCondition() {
         }
 
+        public boolean shouldCascade(DatabaseMapping mapping){
+            return !shouldNotCascade(mapping);
```

Comment:
the whole point of my comment was to avoid negation of negated value..., the idea is for this to be:
```
return shouldCascadeAllParts() || (shouldCascadePrivateParts() && mapping.isPrivateOwned())
```

and `shouldNotCascade` to be:
```
return !shouldCascade(...);
```

so in the end one can avoid having a call stack evaluated to:
`(!(!(shouldCascadeAllParts() || (shouldCascadePrivateParts() && mapping.isPrivateOwned())))`

Reply:
Uff fixed. Needed update `UnitOfWorkImpl` too.